query-planner-js/src/FieldSet.ts

```diff
@@ -21,13 +21,49 @@ export interface Field<
   fieldDef: GraphQLField<any, any>;
 }
 
+/**
+ * Provides a string representation of a field suitable for debugging.
+ *
+ * The format looks like '(a: Int)<A [A1, A2]>' where 'a' is the field name, 'Int' is its type, and '<...>' is its
+ * scope (@see debugScope for details).
+ *
+ * @param field - the field object to convert.
+ * @return a string representation of the field.
+ */
+export function debugField(field: Field) : string {
+  const def = field.fieldDef;
+  return `(${def.name}: ${def.type})${debugScope(field.scope)}`;
+}
+
 export interface Scope<TParent extends GraphQLCompositeType> {
   parentType: TParent;
   possibleTypes: ReadonlyArray<GraphQLObjectType>;
   directives?: ReadonlyArray<DirectiveNode>
   enclosingScope?: Scope<GraphQLCompositeType>;
 }
 
+/**
+ * Provides a string representation of a field suitable for debugging.
+ *
+ * The format looks like '<A [A1, A2]>' where 'A' is the scope 'parentType' and '[A1, A2]' are the 'possibleTypes'.
+ *
+ * @param scope - the scope object to convert.
+ * @param deepDebug - whether to also display enclosed scopes.
+ * @return a string representation of the scope.
+ */
+export function debugScope<TParent extends GraphQLCompositeType>(
+  scope: Scope<TParent>, deepDebug: boolean = false) : string {
+  let enclosingStr = '';
```

Comment:
This doesn't matter at all in practice, but I think it shouldn't be necessary to always initialize `enclosingStr` to an empty string as long as it is initialized somewhere (on all paths). So here that would mean adding an `else` to `if (scope.enclosingScope)` and initializing it to an empty string there. (It could also be that the TypeScript compiler isn't smart enough to analyze the flow with nested conditionals? I haven't tried this out.)

Reply:
Tbh, I kind of like it as it is (but it does work with no initialisation and a `else`). Within reason, I usually feel limiting the amount of flow control you have to go through when reading code improves readability a bit, but I reckon this is probably more personal preference than objective science. So I haven't changed it, but I definitively don't care strongly either so let me know if it bugs you as is and I'll change.

Reply:
That reasoning makes sense, I agree it makes it more readable.